src/components/PeriodDimension/FixedPeriodSelect.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from 'react'
+import propTypes from '@dhis2/prop-types'
+import i18n from '@dhis2/d2-i18n'
+import { SingleSelectField, SingleSelectOption } from '@dhis2/ui'
+import FixedPeriodFilter from './FixedPeriodFilter.js'
+import {
+    parsePeriodCode,
+    getFixedPeriodsOptionsById as getPeriodById,
+    getYearOffsetFromNow,
+} from './utils/fixedPeriods.js'
+import styles from './styles/FixedPeriodSelect.style.js'
+
+const FixedPeriodSelect = ({
+    onChange,
+    allowedPeriodTypes,
+    className,
+    dataTest,
+    value,
+}) => {
+    const [periodType, setPeriodType] = useState('')
+    const [year, setYear] = useState('')
+    const [options, setOptions] = useState()
+
+    useEffect(() => {
+        const period = parsePeriodCode(value, allowedPeriodTypes)
+        if (period) {
+            setPeriodType(period.id)
+            setYear(period.year)
+            setOptions(period.options)
+        } else if (periodType && year) {
+            const offset = getYearOffsetFromNow(year)
+            const options = getPeriodById(periodType).getPeriods({ offset })
+            setOptions(options)
+        }
+    }, [allowedPeriodTypes, periodType, value, year])
+
+    return (
+        <div data-test={dataTest} className={className}>
+            <div className="row">
+                <FixedPeriodFilter
+                    allowedPeriodTypes={allowedPeriodTypes}
+                    currentPeriodType={periodType}
+                    currentYear={year}
+                    onSelectPeriodType={type => {
+                        setPeriodType(type)
+                        onChange('')
```

Comment:
Should this be `null` or `undefined` instead of an empty string...?

Reply:
I could change this, but have done this intentionally. Currently I have defined a default value for `value` to be an empty string in `FixedPeriodSelect.defaultProps`, and when resetting I also chose that value. This way the `value` prop is **always** of type `string`. This has two advantages:
- When I pass the value to `parsePeriodCode` I can safely assume it's a string so I can unconditionally call the `.match()` method.
- _[less relevant]_ I envisage using this component with a form library and typically these will translate `null` or `undefined` to empty strings as well.

Let me know what you prefer.

Reply:
I've discussed this with @martinkrulltott and we decided to go with an implicit `undefined`, which makes the component fall back on its `value` from `defaultProps`. https://github.com/dhis2/analytics/pull/509/commits/c32bf54b1b410a1a6c33ae6dcdff41995344d333